mesonbuild/wrap/wrap.py

```diff
@@ -159,6 +162,10 @@ def parse_wrap_section(self, config: configparser.ConfigParser) -> None:
             raise WrapException(f'{self.wrap_section!r} is not a valid first section in {self.basename}')
         self.type = self.wrap_section[5:]
         self.values = dict(config[self.wrap_section])
+        if 'diff_files' in self.values:
+            FeatureNew('Wrap files with diff_files', '0.61.0').use(self.subproject)
+            diff_files = [s.strip() for s in self.values['diff_files'].split(',')]
+            self.diff_files += diff_files
```

Comment:
This should validate that the patch file does not a) have an absolute path and b) have a `..` segment in it to ensure that only files inside the proper directory are considered. (people have a tendency to do nasty workarounds like these if you let them)

Reply:
Done, I added a check for both cases.